get_fedora_url.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# This script is intended to be called by the Makefile, not by
+# humans.  This implies certain otherwise "odd" behaviors, such
+# as exiting with no std-output if there was an error.  It expects
+# to be called with three arguments:
+# 1. The type of url to retrieve, `image` or `checksum`.
+# 2. The architecture, `x86_64` or `aarch64`
+# 3. The Fedora release, 'rawhide' or a release number.
+
+set -eo pipefail
+
+URL_BASE="https://dl.fedoraproject.org/pub/fedora/linux/"
+CURL="curl --location --silent --fail"
+
+url_type="$1"
+arch_name="$2"
+fed_rel="$3"
+
+die() { echo "ERROR: ${1:-No error message provided}" > /dev/stderr; exit 1; }
+
+msg() { echo "${1:-No error message provided}" > /dev/stderr; }
+
+[[ "$#" -eq 3 ]] || \
+    die "Expecting exactly 3 arguments: <image|checksum> <x86_64|aarch64> <release #>"
+
+tmpfile=$(mktemp -p '' tmp.$(basename ${BASH_SOURCE[0]}).XXXX)
+
+unset output_url
+handle_exit() {
+    local msg_suffix="$arch_name Fedora $fed_rel release $url_type URL."
+    set +e
+    rm -f "$tmpfile" &> /dev/null
+    if [[ -n "$output_url" ]]; then
+        # msg "Found $msg_suffix"
+        echo -n "$output_url" > /dev/stdout
```

Comment:
So I was slightly concerned about the `make` behavior here and if it would slurp up the `\n` or not.  I didn't bother checking this out of pure laziness, but I can spend some time on it today.